google-cloud-pubsub/lib/google/cloud/pubsub/subscription.rb

```diff
@@ -303,6 +303,57 @@ def push_config
           config.freeze
         end
 
+        ##
+        # Inspect the Subscription's bigquery configuration settings. The
+        # configuration can be changed by modifying the values in the method's
+        # block.
+        #
+        #
+        # @yield [bigquery_config] a block for modifying the bigquery configuration
+        # @yieldparam [Google::Cloud::PubSub::V1::BigQueryConfig] bigquery_config
+        #
+        # @return [Google::Cloud::PubSub::V1::BigQueryConfig]
+        #
+        # @example
+        #   require "google/cloud/pubsub"
+        #
+        #   pubsub = Google::Cloud::PubSub.new
+        #
+        #   sub = pubsub.subscription "my-topic-sub"
+        #   sub.bigquery_config.table #=> "my-project:dataset-id.table-id"
+        #   sub.bigquery_config.use_topic_schema #=> true
+        #   sub.bigquery_config.write_metadata #=> false
+        #
+        # @example Update the bigquery configuration by passing a block:
+        #   require "google/cloud/pubsub"
+        #
+        #   pubsub = Google::Cloud::PubSub.new
+        #   sub = pubsub.subscription "my-subscription"
+        #
+        #   sub.bigquery_config do |bc|
+        #     bc.write_metadata = true
+        #     bc.use_topic_schema = false
+        #   end
+        #
+        def bigquery_config
+          ensure_service!
+
+          config = reference? ? nil : @grpc.bigquery_config
+
+          if block_given?
+            old_config = config.dup
+            yield config
```

Comment:
What if `config` is `nil` (from line 341)? Should we still yield it to the block?

Reply:
I have updated to empty Bigquery_config object as how we have it in push_config currently...so user can update accordingly